python/packages/sdk/serverless_sdk/tests/test_trace_span.py

```diff
@@ -1,152 +1,196 @@
-from __future__ import annotations
-from typing_extensions import Final, TypeAlias
-from types import MethodType
-
 import pytest
+import time
+import json
+from serverless_sdk.span.trace import TraceSpan
 
-from . import get_params
-
-
-TEST_NAME: Final[str] = "test.span"
-TEST_INPUT: Final[str] = "Test Input"
-TEST_OUTPUT: Final[str] = "Test Output"
-TEST_START_TIME: Final[int] = 1_000_000
 
+# root span that lives throughout the test session
+@pytest.fixture(scope="session")
+def root_span():
+    span = TraceSpan("test")
+    yield span
+    span.close()
 
-TraceSpan: TypeAlias = "TraceSpan"
 
+def test_root_span(root_span: TraceSpan):
+    assert type(root_span.trace_id) == str, "should automatically generate `trace_id`"
+    assert type(root_span.id) == str, "should automatically generate `id`"
+    assert (
+        type(root_span.start_time) == int
+    ), "should automatically generate `start_time`"
 
-@pytest.fixture
-def trace_span() -> TraceSpan:
-    return get_trace_span()
 
+def test_sub_span(root_span: TraceSpan):
+    # given
+    child_span = TraceSpan("child")
 
-def get_trace_span():
-    from ..span.trace import TraceSpan
+    # when
+    child_span.close()
 
-    return TraceSpan(
-        name=TEST_NAME,
-        input=TEST_INPUT,
-        output=TEST_OUTPUT,
-        start_time=TEST_START_TIME,
+    # then
+    assert (
+        child_span.trace_id == root_span.trace_id
+    ), "should expose `trace_id` of a root span"
+    assert type(child_span.id) == str, "should automatically generate `id`"
+    assert child_span.id != root_span.id, "should have a unique `id`"
+    assert (
+        type(child_span.start_time) == int
+    ), "should automatically generate `start_time`"
+    assert (
+        child_span.start_time != root_span.start_time
+    ), "should have a different `start_time`"
+    assert child_span.name == "child", "should expose `name`"
```

Comment:
Do we have a test that confirms that `parent` span is attributed correctly (and that created spans ends in its child spans) ?

Reply:
Can you elaborate a bit more please 🙏  What do you mean by `parent` span being attributed correctly? And also this bit: 
> created spans ends in its child spans

Reply:
When `child_span` is created, we assign a _parent_ to it: https://github.com/serverless/console/blob/052af60e4979974e30386f5808964a210add38d2/python/packages/sdk/serverless_sdk/span/trace.py#L128
So do we confirm in tests that expected _span_ is assigned as `child_span.parent_span` ?

Also do we confirm that `child_span` ends in `parent_span.sub_spans` collection?

Reply:
Ah I see, yeah good point, I've added these assertions 👍